demo.wiretapp/demo.wiretapp/Wiretapp/WiretappRecordURLProtocol.swift

```diff
@@ -8,12 +9,18 @@ public class WiretappRecordURLProtocol: URLProtocol {
     }
 
     public override class func canInit(with request: URLRequest) -> Bool {
-        return true
+        if let
+            recordingEnabled = ProcessInfo.processInfo.environment[Wiretapp.recordEnabled],
+            recordingEnabled == "true"
+        {
+            return true
+        }
+        return false
     }
 
     public override func startLoading() {
-        self.
-        URLSession.shared.dataTask(with: request) { [weak self] data, response, error in
+        URLSession(configuration: .ephemeral).dataTask(with: request) { [weak self] data, response, error in
```

Comment:
Needed to use a different instance of `URLSession` because `URLSession.shared` was causing recursive calls to these registered implementations